src/sentry/hybridcloud/outbox/category.py

```diff
@@ -0,0 +1,359 @@
+    assert not inter, f"OutboxCategories {inter} were already registered to a different scope"
+    _used_categories.update(categories)
+    return enum_value
+
+
+class OutboxScope(IntEnum):
+    ORGANIZATION_SCOPE = scope_categories(
+        0,
+        {
+            OutboxCategory.ORGANIZATION_MEMBER_UPDATE,
+            OutboxCategory.MARK_INVALID_SSO,
+            OutboxCategory.RESET_IDP_FLAGS,
+            OutboxCategory.ORGANIZATION_UPDATE,
+            OutboxCategory.PROJECT_UPDATE,
+            OutboxCategory.ORGANIZATION_INTEGRATION_UPDATE,
+            OutboxCategory.SEND_SIGNAL,
+            OutboxCategory.ORGAUTHTOKEN_UPDATE_USED,
+            OutboxCategory.POST_ORGANIZATION_PROVISION,
+            OutboxCategory.DISABLE_AUTH_PROVIDER,
+            OutboxCategory.ORGANIZATION_MAPPING_CUSTOMER_ID_UPDATE,
+            OutboxCategory.TEAM_UPDATE,
+            OutboxCategory.AUTH_PROVIDER_UPDATE,
+            OutboxCategory.ORGANIZATION_MEMBER_TEAM_UPDATE,
+            OutboxCategory.API_KEY_UPDATE,
+            OutboxCategory.ORGANIZATION_SLUG_RESERVATION_UPDATE,
+            OutboxCategory.ORG_AUTH_TOKEN_UPDATE,
+            OutboxCategory.PARTNER_ACCOUNT_UPDATE,
+            OutboxCategory.UNUSED_FOUR,
+            OutboxCategory.ISSUE_COMMENT_UPDATE,
+        },
+    )
+    USER_SCOPE = scope_categories(
+        1,
+        {
+            OutboxCategory.USER_UPDATE,
+            OutboxCategory.API_TOKEN_UPDATE,
+            OutboxCategory.UNUSED_ONE,
+            OutboxCategory.UNUSED_TWO,
+            OutboxCategory.UNUSUED_THREE,
+            OutboxCategory.AUTH_IDENTITY_UPDATE,
+        },
+    )
+    WEBHOOK_SCOPE = scope_categories(2, {OutboxCategory.WEBHOOK_PROXY})
+    AUDIT_LOG_SCOPE = scope_categories(3, {OutboxCategory.AUDIT_LOG_EVENT})
+    USER_IP_SCOPE = scope_categories(
+        4,
+        {
+            OutboxCategory.USER_IP_EVENT,
+        },
+    )
+    INTEGRATION_SCOPE = scope_categories(
+        5,
+        {OutboxCategory.INTEGRATION_UPDATE, OutboxCategory.EXTERNAL_ACTOR_UPDATE},
+    )
+    APP_SCOPE = scope_categories(
+        6,
+        {
+            OutboxCategory.API_APPLICATION_UPDATE,
+            OutboxCategory.SENTRY_APP_INSTALLATION_UPDATE,
+            OutboxCategory.SENTRY_APP_UPDATE,
+        },
+    )
+    # Deprecate?
+    TEAM_SCOPE = scope_categories(
+        7,
+        set(),
+    )
+    PROVISION_SCOPE = scope_categories(
+        8,
+        {
+            OutboxCategory.PROVISION_ORGANIZATION,
+        },
+    )
+    SUBSCRIPTION_SCOPE = scope_categories(9, {OutboxCategory.SUBSCRIPTION_UPDATE})
+    RELOCATION_SCOPE = scope_categories(
+        10, {OutboxCategory.RELOCATION_EXPORT_REQUEST, OutboxCategory.RELOCATION_EXPORT_REPLY}
+    )
+
+    def __str__(self) -> str:
+        return self.name
+
+    @classmethod
+    def scope_has_category(cls, shard_scope: int, category: int) -> bool:
+        return OutboxCategory(category) in _outbox_categories_for_scope[shard_scope]
```

Comment:
This method is new and was added to avoid importing `_outbox_categories_for_scope` outside of this module.